qiskit/pulse/instructions/delay.py

```diff
@@ -45,11 +46,16 @@ def __init__(self, duration: int,
             channel: The channel that will have the delay.
             name: Name of the delay for display purposes.
         """
-        super().__init__((duration, channel), duration, (channel,), name=name)
+        super().__init__((duration, channel), None, (channel,), name=name)
 
     @property
     def channel(self) -> Channel:
         """Return the :py:class:`~qiskit.pulse.channels.Channel` that this instruction is
         scheduled on.
         """
         return self.operands[1]
+
+    @property
+    def duration(self) -> int:
```

Comment:
Are you sure about the return type here? The `__init__` method gets `(duration, channel)` as `operands`. So if `def property` returns `self.operands[0]` should the return type not be `Union[int, ParameterExpression]`? The same/ comment applies to the other files.

Reply:
oh you're right. I'll fix